tanh_controller:
  ros__parameters:
    topics:
      vehicle_odometry: /fmu/out/vehicle_odometry
      trajectory_setpoint: /tanh_controller/trajectory_setpoint
      actuator_motors: /fmu/in/actuator_motors
      offboard_control_mode: /fmu/in/offboard_control_mode
      vehicle_command: /fmu/in/vehicle_command
      vehicle_thrust_setpoint: /fmu/in/vehicle_thrust_setpoint

    control_rate_hz: 2000.0
    publish_offboard_control_mode: true
    publish_vehicle_thrust_setpoint: true
    auto_offboard: true
    auto_arm: true
    offboard_warmup: 10

    model:
      mass: 2.0643076923076915
      gravity: 9.81
      inertia_diag: [0.02384669, 0.02394962, 0.04399995]

    position:
      M_P: [2.0, 2.0, 2.0]
      K_P: [1.0, 1.0, 3.0]
      M_V: [5.0, 5.0, 5.0]
      K_V: [0.5, 0.5, 0.5]
      K_Acceleration: [0.0, 0.0, 1000.2]
      max_tilt_deg: 35.0
      observer:
        P_V: [0.1, 0.1, 2.0]
        L_V: [2.0, 2.0, 2.0]

    attitude:
      M_Angle: [2.0, 2.0, 2.0]
      K_Angle: [4.0, 4.0, 4.0]
      M_AngularVelocity: [5.0, 5.0, 5.0]
      K_AngularVelocity: [6.2, 6.2, 5.2]
      K_AngularAcceleration: [0.0, 0.0, 0.0]
      observer:
        P_AngularVelocity: [0.0, 0.0, 0.0]
        L_AngularVelocity: [1.0, 1.0, 1.0]

    allocation:
      l: 0.246073
      beta: 0.78539816339
      cq_ct: 0.016

    motor:
      force_max: 8.54858
      output_map: [1, 3, 0, 2]   # 输出索引0..3 <- 内部索引(默认G1列)：[前左, 前右, 后右, 后左]
